src/kubeseal_auto/core/host.py

```
@@ -203,19 +203,13 @@ def _safe_extract_kubeseal(self, tar: tarfile.TarFile, version: str) -> None:
-            # Verify the extraction path stays within bin_location
-            extract_path = (self.bin_location / target_name).resolve()
-            bin_location_real = self.bin_location.resolve()
-            if not extract_path.is_relative_to(bin_location_real):
-                raise ValueError(f"Path traversal detected: {extract_path}")
-
-            tar.extract(member, path=self.bin_location)
+        # Verify the extraction path stays within bin_location to prevent path traversal
+        extract_path = (self.bin_location / target_name).resolve()
+        bin_location_real = self.bin_location.resolve()
+        if not extract_path.is_relative_to(bin_location_real):
+            raise ValueError(f"Path traversal detected: {extract_path}")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)